### DISCUSSION TITLE
lsusb fails on MacOS Monterey - Intel

### DISCUSSION THREAD

**Discussion by azric** [Homebrew/discussions#3272] [Casks]
I did a "brew install lsusb", which completed, but "lsusb" errors with "invalid integer constant"
```
$ lsusb
2022-05-15 09:43:57.561 system_profiler[38965:846457] SPUSBDevice: IOCreatePlugInInterfaceForService failed 0xe00002be
2022-05-15 09:43:57.562 system_profiler[38965:846457] SPUSBDevice: IOCreatePlugInInterfaceForService failed 0xe00002be
Bus 020 Device 003: ID 05ac:0265 Apple Inc. Magic Trackpad 2  Serial: CC2017200MKJ17LAQ
Bus 020 Device 001: ID 05ac:026c Apple Inc. Magic Keyboard with Numeric Keypad  Serial: F0T02640107HTCYAN
Bus 020 Device 002: ID 05e3:0608 Genesys Logic, Inc. USB2.0 Hub
Bus 020 Device 006: ID 046d:c041 Logitech Inc. USB Gaming Mouse
Bus 020 Device 005: ID 046d:082c Logitech Inc. HD Webcam C615  Serial: 2A2A4390
Bus 020 Device 004: ID 051d:0002 American Power Conversion Back-UPS NS 650M1 FW:929.a5 .D USB FW:a5  Serial: 4B1637P30711
/usr/local/bin/lsusb: line 89: 16#80
80: syntax error in expression (error token is "80")
Bus 000 Device 000: ID 05ac
05ac:8103
8233 Apple Inc.
Apple Inc. Headset  Serial: 000000000000
0000000000000000
/usr/local/bin/lsusb: line 89: 16#: invalid integer constant (error token is "16#")
Bus 000 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
/usr/local/bin/lsusb: line 89: 16#: invalid integer constant (error token is "16#")
Bus 000 Device 000: ID 8086
8086
05ac:8103 8086
8086
Apple Inc. USB 3.1 Bus
```
Anyone have thoughts on how to fix this?

**Comment by bartoszcisek** [comment]
This happen also on M1 Monterey 12.4

**Reply by SMillerDev** [reply]
Have you tried reporting this to the developers instead of the packagers?

**Comment by 3v6l966rd** [comment]
Any solution? I have same issue on OSX Monterey with M1

**Comment by laclouis5** [comment]
I have the same issue. I checked the Homebrew lsusb repo but it has not been updated in 9 years so the project is likely abandoned.

**Comment by chenrui333** [comment]
The issue is actually with `system_profiler SPUSBDataType`, there is no `Bus Number` to parse out (for the below line):

```
bus_num=`echo "$device" | egrep "Bus Number" | sed -e 's/Bus Number://' -e 's/0x//; s/^ *//g' -e 's/ *$//g'`
```

This is the `device` object

```
    USB 3.1 Bus:

      Host Controller Driver: AppleT6000USBXHCI

    USB 3.1 Bus:

      Host Controller Driver: AppleT6000USBXHCI
```

**Comment by phuclv90** [comment]
I've fixed the parsing issue in https://github.com/jlhonora/lsusb/pull/23, you can pull that file and use

**Reply by udp1024** [reply]
It doesn't look like you have merged the pull. The file is still showing from 2014.

**Reply by phuclv90** [reply]
@udp1024 you have to switch to the branch https://github.com/phuclv90/lsusb/tree/fix_broken_lsusb to see the new file

**Comment by frakman1** [comment]
Is this fixed yet? I am on Homebrew 4.3.5 on Sonoma and getting this error:
```
 /opt/homebrew/bin/lsusb: line 89: 16#: invalid integer constant (error token is "16#")
```

**Reply by phuclv90** [reply]
Look at the original repo https://github.com/jlhonora/lsusb and you'll see that it's been abandoned completely 9 years ago. No PR has been merged ever since then. You can use various patches available but they're all just temporary workaround because the original script is already broken in lots of ways and even after fixing just works for the basic cases. Use https://github.com/tuna-f1sh/cyme instead which is far better. You can use `alias lsusb='cyme --lsusb'` to get a `lsusb` compatible output

**Reply by shayneoneill** [reply]
Its updated now. I dont think it was abandoned, it just didnt need to be updated until recently (which has been done).

Also, that cyme thing is pretty great! Thanks for the recomendation.

**Reply by phuclv90** [reply]
nope, it never worked in newer macOS for years so it definitely need an update long long ago. And I looked at the new changes, there are only 2 commits after 10 years, and they're still just some patch work on the existing code. This really needs a refactor. If I were the reviewer the PR won't be approved

**Comment by alfwatt** [comment]
`brew remove lsusb`

`brew install usbutil`

- worked for me

**Comment by shayneoneill** [comment]
Any updates on this. Just did the uninstall/install thing and it still comes back with the outdated version. When I manually install it from the github it works, but it seems I cant convince brew to get the latest version?

**Reply by p-linnane** [reply]
@shayneoneill You need to be more clear about what you're asking. What formula are you referring to when you say "it"?
